packages/flame/lib/src/components/input/joystick_component.dart

```diff
@@ -57,27 +57,36 @@ class JoystickComponent extends HudMarginComponent with Draggable {
           'Either size or background must be defined',
         ),
         assert(
-          knob.position.isZero() && (background?.position.isZero() ?? true),
+          (knob?.position.isZero() ?? true) &&
+              (background?.position.isZero() ?? true),
           'Positions should not be set for the knob or the background',
         ),
         super(
           margin: margin,
           position: position,
-          size: background?.size ?? Vector2.all(size ?? 0),
+          size: background?.size ?? Vector2.zero(),
```

Comment:
This needs to take the `size` variable into account, so it can be changed back to what it was before the PR.

Reply:
Agree, changed.